currency/code.go

```diff
@@ -282,24 +269,26 @@ func (c Code) String() string {
 	if c.Item == nil {
 		return ""
 	}
-	if !c.Item.CaseSensitive || c.upperCase {
+	if c.upperCase {
 		return c.Item.Symbol
 	}
 	return c.Item.Lower
 }
 
-// Lower converts the code to lowercase formatting
+// Lower flags the Code to use LowerCase formatting, but does not change Symbol
+// If Code cannot be lowercased then it will return Code unchanged
 func (c Code) Lower() Code {
-	if c.Item != nil && !c.Item.CaseSensitive {
+	if c.Item == nil || c.Item.Symbol == c.Item.Lower {
```

Comment:
🚨 I believe this might introduce a bug.
```
c := Code{Item: Item{ Symbol: "lower", Lower: "lower": Role: DanceLikeA🐧 }}
c = c.Upper() // Now upperCase is true
c = c.Lower() // Now upperCase is still true
```
I don't see any downside in just doing:
```suggestion:
if c.Item == nil {
 return c
 }
 c.upperCase = false
```
I think it's probably faster to set a bool to false than to check equality.

Reply:
I have purged